File: Leet Code Solutions/1918-maximum-score-of-a-good-subarray/1918-maximum-score-of-a-good-subarray.cs
public class Solution {
    public int MaximumScore(int[] nums, int k) {
        int left = k, right = k;
        int min_val = nums[k];
        int max_score = min_val;

        while (left > 0 || right < nums.Length - 1) {
            if (left == 0 || (right < nums.Length - 1 && nums[right + 1] > nums[left - 1])) {
                right++;
            } else {
                left--;
            }
            min_val = Math.Min(min_val, Math.Min(nums[left], nums[right]));
            max_score = Math.Max(max_score, min_val * (right - left + 1));
        }
        
        return max_score;
    }
}

File: Leet Code Solutions/1918-maximum-score-of-a-good-subarray/1918-maximum-score-of-a-good-subarray.py
class Solution:
    def maximumScore(self, nums: List[int], k: int) -> int:
        left, right = k, k
        min_val = nums[k]
        max_score = min_val
        
        while left > 0 or right < len(nums) - 1:
            if left == 0 or (right < len(nums) - 1 and nums[right + 1] > nums[left - 1]):
                right += 1
            else:
                left -= 1
            min_val = min(min_val, nums[left], nums[right])
            max_score = max(max_score, min_val * (right - left + 1))
        
        return max_score